### DISCUSSION TITLE
antmedia with turn server not work

### DISCUSSION THREAD

**Discussion by benzidane027** [ant-media/Ant-Media-Server#6774] [General]
### Short description
__Brief description of what happened__
  hell,

 i installed ams ce in my local machine  and i installed cournt in amazone ec2 server i check my cournt server with this site :
<a href="https://webrtc.github.io/samples/src/content/peerconnection/trickle-ice/" > here <a/>

reasul was :

Time	Type	Foundation	Protocol	Address	Port	Priority	URL (if present)	relayProtocol (if present)
0.008	host	1424801752	udp	11c87c77-14f3-4073-b9b1-f2017fe10452.local	44907	126 | 30 | 255		
2.403	srflx	2295364930	udp	x.x.x.x	25813	100 | 30 | 255	stun:y.y.y.y:3478	
4.378	relay	1905995084	udp	y.y.y.y	58855	2 | 31 | 255	turn:y.y.y.y:3478?transport=udp	udp
4.379	Done

my coturn conf:

realm=myDomain.com     
server-name=live-turn
listening-ip=0.0.0.0
external-ip=y.y.y.y
listening-port=3478
fingerprint
log-file=/var/log/turnserver.log
user=user:pass
lt-cred-mech
tls-listening-port=5349

my ams config:
  "stunServerURI": "turn:y.y.y.y?transport=udp",
  "turnServerUsername": "user",
  "turnServerCredential": "pass",

javascript :
    var pc_config = {
      iceServers: [

        {
          username: "user",
          credential: "pass",
          urls: ["turn:y.y.y.y:3478?transport=udp"],
        },
      ],
    };

    var sdpConstraints = {
      OfferToReceiveAudio: true,
      OfferToReceiveVideo: true,
    };

    let webrtcAdaptor = new WebRTCAdaptor({
      websocket_url: WsSocket,
      localVideoElement: sourceCallerCam,
      pc_config: pc_config,
      mediaConstraints: mediaConstraints,
      sdpConstraints: sdpConstraints,
      candidateTypes: ["tcp", "udp"],
      remoteVideoId: "remoteVideo",

      callback: (info, obj) => {
        if (info === "publish_started") {
          alert("Publish started!");
        } else if (info === "publish_finished") {
          console.log("publish finished");
        } else if (info === "initialized") {
          console.log("WebRTC initialized");
          webrtcAdaptor.publish(streamId);
        }
      },
    });
  };

### Environment

* Operating system and version: ubuntu 20.04
* Java version: 
* Ant Media Server version: Community Edition 2.11.3 20240906_1141
* Browser name and version: firefox , brave, chrom

### Steps to reproduce
1. 
2. 
3.

### Expected behavior
__Put as much detail here as possible__
 start streaming from my browser to ant media server 
### Actual behavior
__Put as much detail here as possible__
  streaming not start

)

**Comment by muratugureminoglu** [comment]
Hi @benzidane027 

If you run a coturn behind NAT, you need to [configure](https://antmedia.io/docs/guides/advanced-usage/turn-instalation/coturn-quick-installation/) it accordingly. 

For example, there should be a line like the following.
```
external-ip=public_ip/private_ip
```
You can easily install it using the script [here](https://github.com/ant-media/Scripts/blob/master/install_turn-server.sh).

Regards.

**Comment by benzidane027** [comment]
thanks for answer bro
i am  using cloudflare turn server i checked on  Trickle ICE and it work 
but in my app i got same result   like before

**Comment by muratugureminoglu** [comment]
Hi @benzidane027 

Let's be on the same page. 

Are you using CloudFlare turn server or on EC2 ? Because as your messages, there are 2 things as you mentined. Cloudflare and your own turn server on AWS. 

I 've just tried Cloudflare turn server on my end, and it 's working properly. 

![image](https://github.com/user-attachments/assets/8e1a8317-1da0-4d12-ad08-1118096271fc)

![image](https://github.com/user-attachments/assets/f68c28c1-82f3-43d1-abcf-058b30cc36e1)


If you are testing on your local network(Lan-to-Lan), it need not to install turn server. If you are trying to reach it from outside of your network. Need to check internet connectivity.

**Comment by benzidane027** [comment]
after some research i find out i can use turn server on cloud flare so :

1- i installed ams on ec2 and i get cloudflare  turn server i do config and start node app to stream with webrtc to ams it did'nt work
2- i install ams in my local machine and install coturn on ec2 instance .  i start node app to stream with webrtc to ams it did'nt work

3- i install ams in my local machine . with google stun server. i start node app to to stream with webrtc to ams and it work good

* my mean goal id  start ams on ec2 with cloudflare turn server

this is log for turn server in cloudflare
i check the server 2 time so: 
![Screenshot from 2024-11-06 12-21-53](https://github.com/user-attachments/assets/1572da17-c8f5-4248-ac95-0b31aeae20c7)

**Comment by benzidane027** [comment]
ams on ec2 conf:
![Screenshot from 2024-11-06 12-26-57](https://github.com/user-attachments/assets/a528231f-8fef-4350-9e89-cc1947add002)

**Comment by benzidane027** [comment]
my javascript code :
![Screenshot from 2024-11-06 12-30-14](https://github.com/user-attachments/assets/bd599467-d9e9-4db2-8de6-3a6b13963d12)

**Comment by benzidane027** [comment]
any suggest i stuck in this step

**Reply by muratugureminoglu** [reply]
You have done a lot of experimenting here and this puts you in a loop. Write down exactly what you want to do. Go one step at a time.

 Why do you need a turn server ?

**Reply by benzidane027** [reply]
well . i want to make stream platform some users can t start webrtc stream from browser to ams server throught  stun server(close network , firewall ...) so i want to add turn server to make everyone can stream in my platform

**Reply by muratugureminoglu** [reply]
If you share your turn server credentials and ams dashboard credentials with me through support[at]antmedia[dot]io, I can check it for you.

**Reply by muratugureminoglu** [reply]
Hi @benzidane027 

I 've just deleted your answer. Because there were credentials.  FYI.

Please don't share credentials in public platform.

**Reply by benzidane027** [reply]
thank you
this is just a test server i will delete it later

**Reply by muratugureminoglu** [reply]
Hi @benzidane027  

Could you please share the Security Groups (Outbound/Inbound) of the instance you are using in AWS?

**Reply by benzidane027** [reply]
![Screenshot From 2024-11-25 09-53-18](https://github.com/user-attachments/assets/832320a1-f2f0-4b6c-b29f-e79b91d50588)

**Reply by muratugureminoglu** [reply]
Let's fix your issue in the meeting. I'll send you a meeting invitation if you're available at 8 pm (GMT +3) today. If it 's ok please share your email address by support[@]antmedia.io

Regards.

**Reply by benzidane027** [reply]
ok . i will contact you 
thank you men

**Reply by benzidane027** [reply]
email sent throughout support account

**Comment by benzidane027** [comment]
turn server in cloudflare not evenly notify about income connections from client or ams

**Comment by benzidane027** [comment]
is that a miss config ??
